src/renderer/sw_engine/tvgSwShape.cpp

```diff
@@ -340,6 +370,8 @@ static SwOutline* _genDashOutline(const RenderShape* rshape, const Matrix* trans
     auto offset = 0.0f;
     dash.cnt = rshape->strokeDash((const float**)&dash.pattern, &offset);
     auto simultaneous = rshape->stroke->trim.simultaneous;
+    float trimBegin, trimEnd;
+    _recalculateTrim(rshape->stroke, trimBegin, trimEnd);
```

Comment:
We can put the main update in the tvgRender.h so that other backend engines can commonly use it.
```
//valid == true, when trimming should work
auto valid = rshape->stroke->trim(trimBegin, trimEnd);  //return the values applyed the looping
```

Reply:
@hermet done. please notice that we have strokeTrim in renderShape and adjustTrim in RenderStroke. 
adjustTrim returns bool like strokeTrim, but I didn;t want to unify them, because not always when we ask about validation we need to adjust the values.